core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1404,12 +1419,14 @@ public function delete($blnDoNotRedirect=false)
 			throw new InternalServerErrorException('Table "' . $this->strTable . '" is not deletable.');
 		}
 
-		if (!$this->intId)
+		$this->loadActiveRecord();
+
+		if (null === $this->activeRecord)
 		{
 			$this->redirect($this->getReferer());
```

Comment:
This should probably throw a 404 exception instead? Since it should never happen, the redirect is not "useful" for the back end.

Reply:
I don't want to change behaviour in this PR.

Reply:
see https://github.com/contao/contao/pull/4826